src/Microsoft.DotNet.Compiler.Common/CommonCompilerOptionsExtensions.cs

```diff
@@ -205,6 +212,11 @@ public static IEnumerable<string> SerializeToArgs(this CommonCompilerOptions opt
                 args.Add(s_generateXmlDocumentation.ToLongArg(generateXmlDocumentation));
             }
 
+            if (outputName != null)
```

Comment:
Empty values are being checked elsewhere. The error message is not great as it does relate exactly to the name from project.json (`option --output-name requires a value`) - but this is consistent with other options (I think I checked the xmlDoc option)